File: src/main/java/com/timepath/quakec/ide/syntax/QCSyntaxHighlighterFactory.java
package com.timepath.quakec.ide.syntax;

import com.intellij.openapi.fileTypes.SingleLazyInstanceSyntaxHighlighterFactory;
import com.intellij.openapi.fileTypes.SyntaxHighlighter;
import org.jetbrains.annotations.NotNull;

/**
 * @author TimePath
 */
public class QCSyntaxHighlighterFactory extends SingleLazyInstanceSyntaxHighlighterFactory {
    @NotNull
    @Override
    public SyntaxHighlighter createHighlighter() {
        return new QCSyntaxHighlighter();
    }
}

File: src/main/kotlin/com/timepath/quakec/ide/syntax/QCSyntaxHighlighterFactory.kt
package com.timepath.quakec.ide.syntax

import com.intellij.openapi.fileTypes.SingleLazyInstanceSyntaxHighlighterFactory

class QCSyntaxHighlighterFactory : SingleLazyInstanceSyntaxHighlighterFactory() {
    override fun createHighlighter() = QCSyntaxHighlighter()
}
